### PR TITLE
Add trait for filtering draft PRs

### PR DESCRIPTION
**Pull request by cwholmes**
# Description

This change introduces a filter trait for draft pull requests. This can be useful for projects that want to limit builds of pull request to only the pull requests that are ready for merge.

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Automated tests have been added to exercise the changes
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

**Comment by agaudreault-jive** [comment]
I created https://issues.jenkins.io/browse/JENKINS-65156 few months ago. You can use this ticket for your PR.

File: src/main/java/org/jenkinsci/plugins/github_branch_source/IgnoreDraftPullRequestFilterTrait.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                return true;
+              }
+              return false;
+            }
+            return false;
+          }
+        });
+  }
+
+  @Symbol({"ignoreDraftPullRequestFilter"})
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/IgnoreDraftPullRequestFilterTrait.java:55]:
```suggestion
  @Symbol({"gitHubIgnoreDraftPullRequestFilter"})
```

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties [hunk 2]
```diff
@@ -71,3 +71,5 @@ GitHubSCMNavigator.general=General
 GitHubSCMNavigator.withinRepository=Within repository
 
 GitHubAppCredentials.displayName=GitHub App
+
+IgnoreDraftPullRequestFilterTrait.DisplayName=Ignore pull requests marked as drafts.
```
Comment (bitwiseman) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/Messages.properties:75]:
```suggestion
IgnoreDraftPullRequestFilterTrait.DisplayName=Ignore pull requests marked as drafts
```

**Review by bitwiseman — APPROVED** [review]
Minor tweak but otherwise looks great.  Have you tested locally on a Jenkins instance?

**Comment by cwholmes** [comment]
> Minor tweak but otherwise looks great. Have you tested locally on a Jenkins instance?

I built and installed the plugin locally and tested a sample repo.

Here is the trait in the list for the repo build:

trait-in-list

The trait added to the list for the repo build:

trait-added

The pull requests in the sample repo with one being a draft:

prs-in-repo

The scan log excluding the draft pr build:

scan-log

The pull request tab for the repo build not including the draft:

pr-builds

The branches tab for the repo build also not including the draft pr branch:

branch-builds

**Comment by KalleOlaviNiemitalo** [comment]
Filed [JENKINS-66503](https://issues.jenkins.io/browse/JENKINS-66503): GitHub "Ignore pull requests marked as drafts" trait is available for non-GitHub branch sources.

**Pull request merged**